runtime/src/message_processor.rs

```diff
@@ -577,6 +587,11 @@ impl MessageProcessor {
             .iter()
             .map(|seeds| Pubkey::create_program_address(&seeds, caller_program_id))
             .collect::<Result<Vec<_>, solana_sdk::pubkey::PubkeyError>>()?;
+        let mut caller_privileges = keyed_accounts
+            .iter()
+            .map(|keyed_account| keyed_account.is_writable())
+            .collect::<Vec<bool>>();
+        caller_privileges.insert(0, false);
```

Comment:
@jackcmay why insert `false` here? And isn't `caller_privileges` supposed to be aligned with `accounts` rather than `keyed_accounts`?

Reply:
IIRC it holds the space of the program, which isn't part of the `keyed_accounts` and the program is not writeable

Reply:
Got it, it doesn't look like that's the case anymore on latest master so I have a fix here: https://github.com/solana-labs/solana/pull/18616